Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs

```diff
@@ -326,6 +340,13 @@ onUpdate ValidatedRideCompletedReq {..} = do
           }
 onUpdate ValidatedBookingCancelledReq {..} = do
   mbRide <- QRide.findActiveByRBId booking.id
+
+  let isBookingCancellable =
+        booking.status `elem` [SRB.NEW, SRB.CONFIRMED, SRB.AWAITING_REASSIGNMENT, SRB.TRIP_ASSIGNED]
+      isRideCancellable = maybe False (\ride -> ride.status `notElem` [SRide.INPROGRESS, SRide.CANCELLED]) mbRide
+      bookingAlreadyCancelled = booking.status == SRB.CANCELLED
+  unless (isBookingCancellable || (bookingAlreadyCancelled && isRideCancellable)) $
```

Comment:
These checks are already there in `validateRequest` function. So could you combine them?

Reply:
Done. Please, check